src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -426,59 +435,137 @@ def next_edges_by_connector(
             )
 
 
-def _gt_map_strides_to_nested_sdfg(
+def _gt_map_strides_into_nested_sdfg(
     nsdfg_node: dace.nodes.NestedSDFG,
     inner_data: str,
-    edge_data: dace.Memlet,
-    outer_strides: Iterable[int | dace.symbolic.SymExpr],
+    outer_subset: dace.subsets.Subset,
+    outer_desc: dace_data.Data,
     ignore_symbol_mapping: bool = False,
 ) -> None:
-    """
+    """Modify the strides of `inner_data` inside `nsdfg_node` to match `outer_desc`.
+
+    `inner_data` is the name of of a data descriptor inside the NestedSDFG.
+    The function will then modify the modify the strides of `inner_data` to
+    match the ones of `outer_desc`.
+
+    Args:
+        nsdfg_node: The node in the parent SDFG that contains the NestedSDFG.
+        inner_data: The name of the data descriptor that should be processed
+            inside the NestedSDFG (by construction also a connector name).
+        outer_subset: The subset that describes what part of the outer data is
+            mapped into the NestedSDFG.
+        outer_desc: The data descriptor of the data on the outside.
+        ignore_symbol_mapping: If possible the function will perform the renaming
+            through the `symbol_mapping` of the nested SDFG. If `True` then
+            the function will always perform the renaming.
+
     Todo:
-        - Refactor this function.
-        - Handle the case the stride is used somewhere else.
-        - Handle the case where we have an explicit size 1 dimension in slicing.
+        - Handle explicit dimensions of size 1.
     """
-    # We need to propagate the strides inside the nested SDFG on the global arrays
-    new_strides = tuple(
-        stride
-        for stride, to_map_size in zip(
-            outer_strides,
-            edge_data.subset.size(),
-            strict=True,
-        )
-        if to_map_size != 1
-    )
-    inner_desc = nsdfg_node.sdfg.arrays[inner_data]
+    # We need to compute the new strides. In the following we assume that the
+    #  relative order of the dimension does not change, but some dimensions
+    #  that are present on the outside are not present on the inside. For
+    #  example this happens for the Memlet `a[__i0, 0:__a_size1]`.
+    #  We detect this case by checking if that dimension has size 1.
+    # TODO(phimuell): Handle the case were some additional size 1 dimensions are added.
+    inner_desc: dace_data.Data = nsdfg_node.sdfg.arrays[inner_data]
+    inner_shape = inner_desc.shape
+    inner_strides_init = inner_desc.strides
+
+    # TODO(phimuell): For now this is fine, but it should be possisble to allow it.
```

Comment:
You are right, we do not need to impose this limitation. You can remove the assert.